client/src/components/EditProfile.tsx

```diff
@@ -25,12 +26,27 @@ export default function EditProfile() {
   const [editLoading, setEditLoading] = useState(false);
   const [error, setError] = useState("");
 
-  const handleSubmit = async () => {
-    const parsedUserProfile = userProfileSchema.safeParse(userProfile);
-    if (!parsedUserProfile.success) {
-      setError("Invalid form.");
+  console.log(userProfile);
+
+  const handleEditProfile = () => {
+    setError("");
+
+    const parsedUserProfile = userProfileSchema.parse(userProfile);
```

Comment:
`.parse` potentially throws an error and rejects the promise, but we aren't handling it. I suggest using `safeParse` or wrapping this in a `try-catch` block. A `try-catch` block is probably cleaner.